sync-android/src/test/java/com/cloudant/sync/datastore/encryption/EndToEndEncryptionTest.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) 2015 IBM Cloudant. All rights reserved.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.datastore.encryption;
+
+import com.cloudant.sync.datastore.Datastore;
+import com.cloudant.sync.datastore.DatastoreManager;
+import com.cloudant.sync.datastore.DatastoreNotCreatedException;
+import com.cloudant.sync.datastore.DocumentBodyFactory;
+import com.cloudant.sync.datastore.DocumentException;
+import com.cloudant.sync.datastore.MutableDocumentRevision;
+import com.cloudant.sync.datastore.UnsavedFileAttachment;
+import com.cloudant.sync.query.IndexManager;
+
+import net.sqlcipher.database.SQLiteDatabase;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.commons.io.IOUtils;
+import org.hamcrest.core.IsEqual;
+import org.hamcrest.core.IsNot;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertThat;
+
+
+/**
+ * Test that when you open a database with an encryption key, the content on disk
+ * is all encrypted.
+ */
+@RunWith(Parameterized.class)
+public class EndToEndEncryptionTest {
+
+    static {
+        // Load SQLCipher libraries
+        SQLiteDatabase.loadLibs(ProviderTestUtil.getContext());
```

Comment:
It's a good point I'd not thought of. However, once we load the libraries into the JVM, we've loaded the libraries for all the tests, presuming they are all run in the same JVM? I'm not sure we gain from using the `@BeforeClass` method. Happy to change it if you think it's worth while.